### DISCUSSION TITLE
Access state outside of react lifecycle?

### DISCUSSION THREAD

**Discussion by Geddard** [pmndrs/jotai#694] [Q&A]
Hi there, I'm wondering if I can somehow read the state of an atom outside of the normal component lifecycle, not using hooks.

With redux for example i would do it like this:

```
import { default as store } from 'store'

store.getState().myState.someValue;

```

I can't figure out or find info about doing something like this with Jotai, is it even possible?

Thanks!

**Comment by dai-shi** [comment]
jotai atoms are modeled after `useState` and you can't access atom values outside of react lifecycle without using hooks.
(we may re-consider this in the future, but for now it's conceptually not possible.)

If you don't want doing it with hooks, (which I totally agree!)
one option is to use `onMount` and subscribe to variables outside react lifecycle.
`jotai/valtio` and `jotai/zustand` (and `jotai/redux` too) provide utils with this technique.

**Reply by Geddard** [reply]
Thanks for the suggestion, but i think i will try to sync the atom with a local storage value and read it from there to make things a bit more "legal" xD

**Comment by dai-shi** [comment]
Newer version of jotai has an experimental feature `unstable_createStore`. See: #861 and #922
Related: #1028, #1029, #1085, #999, and maybe some more.

(Noticed this discussion from the @rodrigocfd 's blog post: https://bluedesk.blogspot.com/2022/06/global-usestate-hooks-with-jotai.html )

**Reply by nrgnrg** [reply]
Thanks for this.

I've been able to use `unstable_createStore` to share atoms and state between multiple islands in Astro to prevent having to hydrate closer to the root of the tree.

I'm also scoping a `store` to a specific request as it comes in on the server for SSR.

Exposing `unstable_createStore` makes so much more possible with this library.
